### DISCUSSION TITLE
Integration with typescript.nvim?

### DISCUSSION THREAD

**Discussion by zeitchef** [VonHeikemen/lsp-zero.nvim#39] [Q&A] [answered]
I was previously using this plugin as part of my daily workflow before switching to `lsp-zero` - I'm wondering how might it be possible to configure `tsserver` with `typescript.nvim`? 

Before I had to do something like this, passing in `tsserver` options:

```
  require('typescript').setup {
    debug = false,
    server = {
      on_attach = function(client, bufnr)
        ...
      end,
    },
  }
```

Thanks!

**Comment by VonHeikemen** [comment, marked as answer]
In this case you should use the function `.build_options()` to create the config for `tsserver` then pass that to typescript.nvim.

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

lsp.skip_server_setup({'tsserver'})

lsp.setup()

require('typescript').setup({
  debug = false,
  server = lsp.build_options('tsserver', {})
})
```

If you need to customize tsserver add your options inside the curly braces. Here is an example.

```lua
lsp.build_options('tsserver', {
  on_attach = function()
    print('hello')
  end,
  settings = {
    completions = {
      completeFunctionCalls = true
    }
  }
})
```

**Reply by zeitchef** [reply]
¡Excelente! ¡Gracias!
